test/regression/regression_test.go

```diff
@@ -0,0 +1,382 @@
+//nolint:dupl
+package regression
+
+import (
+	"context"
+	"fmt"
+	"net/http"
+	"os"
+	"os/exec"
+	"runtime"
+	"testing"
+	"time"
+
+	"github.com/drand/drand/v2/common/log"
+	"github.com/drand/drand/v2/internal/chain"
+	"github.com/drand/drand/v2/internal/core"
+	"github.com/drand/drand/v2/internal/dkg"
+	"github.com/drand/drand/v2/internal/net"
+	"github.com/drand/drand/v2/internal/test"
+	dkgproto "github.com/drand/drand/v2/protobuf/dkg"
+	"github.com/drand/drand/v2/protobuf/drand"
+	"github.com/jonboulle/clockwork"
+	"github.com/stretchr/testify/require"
+)
+
+var oldBinaryPath string
+
+func init() {
+	switch runtime.GOOS {
+	case "linux":
+		oldBinaryPath = "./drand-1.5.7-linux"
```

Comment:
Why 1.5.7 specifically?

Reply:
Iirc it was the version that IPFS force were running on when they had issues - I think v1.5.8 contained some migration paths